presto-main/src/main/java/com/facebook/presto/operator/aggregation/OddsRatioAggregation.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.operator.aggregation.state.OddsRatioState;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import io.airlift.slice.Slice;
+
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.google.common.base.Preconditions.checkArgument;
+
+@AggregationFunction
+public final class OddsRatioAggregation
+{
+    private OddsRatioAggregation() {}
+
+    /**
+     * odds_ratio(input_col, input_val1, input_val2, output_col, output_val)
+     * -> {low_bound: lb , odds_ratio: or , upper_bound:ub}
+     * <p>
+     * Calculates the odds ratio of input_val1 to input_val2 given the output_val.
+     * Calculates the ratio of the odds of output_val occurring given input_val1 to the odds of output_val occurring given input_val2.
+     */
+    @InputFunction
+    public static void input(
+            @AggregationState OddsRatioState state,
+            @SqlType(StandardTypes.VARCHAR) Slice input_col,
+            @SqlType(StandardTypes.VARCHAR) Slice input_val1,
+            @SqlType(StandardTypes.VARCHAR) Slice input_val2,
+            @SqlType(StandardTypes.VARCHAR) Slice output_col,
+            @SqlType(StandardTypes.VARCHAR) Slice output_val)
+    {
+        state.setTotalCount(state.getTotalCount() + 1);
+        if (output_val.equals(output_col)) {
+            if (input_col.equals(input_val1)) {
+                state.setInput1Count(state.getInput1Count() + 1);
+            }
+            else if (input_col.equals(input_val2)) {
+                state.setInput2Count(state.getInput2Count() + 1);
+            }
+        }
+    }
+
+    @CombineFunction
+    public static void combine(@AggregationState OddsRatioState state, @AggregationState OddsRatioState otherState)
+    {
+        long otherCount = otherState.getTotalCount();
+        checkArgument(otherCount >= 0, "count is negative");
```

Comment:
❓ I copied this check from other state merging functions, but I was curious how this condition could arise in practice. Is this check necessary?